pkg/controller/vitesscell/reconcile_vtgate.go

```diff
@@ -116,6 +116,12 @@ func (r *ReconcileVitessCell) reconcileVtgate(ctx context.Context, vtc *planetsc
 	}
 	update.Annotations(&annotations, vtc.Spec.Gateway.Annotations)
 
+	// Merge ExtraVitessFlags into ExtraFlags.
```

Comment:
I think we should merge in the opposite order, so the more-specific flags (for vtgate) take precedence over less-specific (vitess-wide) flags.

Also, we should create a new, empty map to avoid mutating our inputs.

Reply:
Done https://github.com/planetscale/vitess-operator/pull/14/commits/1a12cb834722dc60a7ce9fbe93dbfa70ba5cc92a.